[nathanshew] iP

nathanshew opened the pull request:
### DukePro

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

- text-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [ ] Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:
***
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

nathanshew committed:
Rename, Greet, Exit

nathanshew committed:
Use Text Block

nathanshew committed:
Rename from Duke to NathanBot

nathanshew committed:
Level 1. Echo
Improve the skeletal version of Duke so that it echos commands entered by the user, and exits when the user types the command bye.

nathanshew committed:
Level 2. Add, List

nathanshew committed:
Level 3. Mark as Done

nathanshew committed:
Level 4. ToDos, Events, Deadlines

Cleanup using CoPilot

nathanshew committed:
Level 5. Handle Errors

nathanshew committed:
give the user a clear/specific explanation of the error and how to correct it.

nathanshew committed:
Level 6. Delete

Clean up white space

nathanshew committed:
Rename Tasklist

nathanshew committed:
Rename tasklist

nathanshew committed:
A-Enums

nathanshew committed:
Refactoring code

nathanshew committed:
Refactor Code

nathanshew committed:
A-TextUiTesting: Automated Text UI Testing

nathanshew committed:
Level 7. Save

nathanshew committed:
Update gitignore

nathanshew committed:
Level 8. Dates and Times

nathanshew committed:
Bugfix, to create directory if it does not already exist

nathanshew committed:
Storage

nathanshew committed:
A-MoreOOP Storage

nathanshew committed:
Fix typo error

nathanshew committed:
Introuduction of UI, parser already previously inbuilt.

nathanshew committed:
Packaging

nathanshew committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

nathanshew committed:
refactoring of code

nathanshew committed:
A-JUnit: Add JUnit Tests

nathanshew committed:
Attributions

nathanshew committed:
A-Jar

nathanshew committed:
A-JavaDoc
↳ Add JavaDoc comments

nathanshew committed:
Cleaned up using Copilot to follow Google's Java Style Guide, while keeping indentations to be 4 spaces.

nathanshew committed:
Academic Declarations

nathanshew committed:
Code refactoring

nathanshew committed:
Level 9. Find

nathanshew committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/nathanbot/commands/CommandHandler.java
#	src/main/java/nathanbot/commands/CommandType.java
#	src/main/java/nathanbot/storage/Storage.java
#	src/main/java/nathanbot/tasks/Deadline.java
#	src/main/java/nathanbot/tasks/Event.java
#	src/main/java/nathanbot/tasks/Task.java
#	src/main/java/nathanbot/tasks/TaskList.java
#	src/main/java/nathanbot/tasks/ToDo.java
#	src/main/java/nathanbot/ui/UserInterface.java

nathanshew committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/nathanbot/ui/UserInterface.java

nathanshew committed:
To follow Java StyleGuide

nathanshew committed:
Attributions

Reallyeasy1 commented on .gitignore:
```diff
@@ -3,6 +3,9 @@
 /*.iml
 
+# Data files
+/data/*
```
Good that you added data in .gitignore

Reallyeasy1 commented on Academic Declarations/JavaCodingStandard2103.txt:
```diff
@@ -0,0 +1,103 @@
+@params can be omitted if all parameters of a method have self-explanatory names or already explained in the main part of the comment i.e., when adding @params to the comment does not add any value. This means the comment will have @param for all its parameters, or none.
+When writing Javadocs for overridden methods, the @inheritDoc tag can be used to reuse the header comment from the parent method but with further modifications e.g., when the method has a slightly different behavior from the parent method.
+Javadoc of class members can be specified on a single line
+Comments should be indented relative to their position in the code.
```
Good that you have added a java coding standard file

Reallyeasy1 commented on src/test/java/nathanbot/tasks/TaskListTest.java:
```diff
@@ -0,0 +1,53 @@
+        String expected = "1. Task 1\n2. Task 2\n";
+        assertEquals(expected, taskList.toString());
+    }
+}
```
Good that you mock the objects for testing

Reallyeasy1 reviewed (commented):
Good overall, just minor changes

zeonchew0324 commented on src/main/java/nathanbot/commands/CommandHandler.java:
```diff
@@ -0,0 +1,208 @@
+            int index = Integer.parseInt(input.substring(CommandType.DELETE.getCommand().length()));
+            Task task = taskList.getTask(index - 1);
+            taskList.deleteTask(index - 1);
+            System.out.println(LINE + "Noted. I've removed this task:\n  " + task + "\nNow you have " + taskList.listLength() + " tasks in the list.\n" + LINE);
```
Wouldn't it be better to make this line into two so that it would not hit the max character of 120?

zeonchew0324 commented on src/main/java/nathanbot/commands/CommandHandler.java:
```diff
@@ -0,0 +1,208 @@
+     * @param taskList The list of tasks.
+     * @param isDone True if the task should be marked as done, false otherwise.
+     */
+    public static void handleMarkCommand(String input, String command, TaskList taskList, boolean isDone) {
```
Good that you followed the boolean naming convention!

zeonchew0324 commented on src/test/java/nathanbot/tasks/DeadlineTest.java:
```diff
@@ -0,0 +1,19 @@
+import org.junit.jupiter.api.Test;
+
+import nathanbot.tasks.Deadline;
+
```
Good consistency of the ordering of import statements!

zeonchew0324 commented on src/main/java/nathanbot/ui/UserInterface.java:
```diff
@@ -0,0 +1,75 @@
+                CommandType commandType = CommandType.fromInput(input);
+
+                switch (commandType) {
+                case BREAK:
```
Followed the correct switch case coding standard, good job!

zeonchew0324 reviewed (commented):
LGTM! Just be cautious that some lines could break into two so that it is not too long for a single line of code.

nathanshew committed:
Add Checkstyle

nathanshew committed:
Update style

nathanshew committed:
Add newline

nathanshew committed:
Implement GUI

nathanshew committed:
Update Javadocs

nathanshew committed:
Fixed formatting issues

nathanshew committed:
Update build.gradle

nathanshew committed:
Classify Files

nathanshew committed:
Add isEmpty()

Improve SLAP by abstracting isEmpty, improving code quality

nathanshew committed:
Add Assertions

Commands can be mistakenly passed into Command Handler.

This causes calculation errors, and returns wrong descriptions.

New Assertions ensure that the commands passed in start with the correct commands.

nathanshew committed:
Merge pull request #1 from nathanshew/branch-A-CodeQuality

Branch a code quality

nathanshew committed:
Merge branch 'master' of https://github.com/nathanshew/ip into branch-A-Assertions

nathanshew committed:
Merge pull request #2 from nathanshew/branch-A-Assertions

Add Assertions

nathanshew committed:
Add Tags to tasks

Current tasks cannot be tagged, which is a problem for users.

Tagging function allows us to add relevant tags to remember the tasks better.

nathanshew committed:
Merge pull request #3 from nathanshew/branch-BCD-Extension

Add Tags to tasks

nathanshew committed:
Fix Formatting

nathanshew committed:
Add AI.md

Edit files to refer to AI.md, rather than Academic Declarations.

nathanshew committed:
Add README.md

Add README, and tidy code.

nathanshew committed:
Update README.md

nathanshew committed:
Update AI.md

nathanshew committed:
Update Storage.java
